### PR TITLE
fixed redis cache and info api endpoint

### PR DESCRIPTION
**Pull request by AinzOwl**

### CODE REVIEW

File: src/routes/anime/gogoanime.ts [hunk 1]
```diff
@@ -45,29 +46,39 @@ const routes = async (fastify: FastifyInstance, options: RegisterOptions) => {
...
+            }
+        };
+
+        // Attempt to retrieve the information, using cache if available.
+        let res = redis ? await cache.fetch(
+            (redis as Redis),
+            `${redisPrefix}info;${id}`,
+            fetchAnimeInfo, // Here you just pass the reference to the function
+            redisCacheTime,
+        ) : await fetchAnimeInfo();
```
Comment (riimuru) [inline src/routes/anime/gogoanime.ts:69]: `fetchAnimeInfo` function isn't needed, you can just use `gogoanime.fetchAnimeInfo(id)`

File: src/routes/anime/gogoanime.ts [hunk 2]
```diff
@@ -45,29 +46,39 @@ const routes = async (fastify: FastifyInstance, options: RegisterOptions) => {
     reply.status(200).send(res);
   });
 
-  fastify.get('/info/:id', async (request: FastifyRequest, reply: FastifyReply) => {
+  fastify.get('/info/:id', async (request, reply) => {
```
Comment (riimuru) [inline src/routes/anime/gogoanime.ts:49]: Is there any reason you removed the typings here?

**Review by riimuru — COMMENTED** [review]
Can you elaborate on the issue that this pr fixes? How was the Redis cache broken on this endpoint?

**Comment by New-dev0** [comment]
> Can you elaborate on the issue that this pr fixes? How was the Redis cache broken on this endpoint?
that endpoint is giving {}  as a response, apply changes from this pr seems to fix the issue

**Review by riimuru — COMMENTED** [review]
#583 closes this pr

**Pull request closed without merging**